metadata-ingestion/src/datahub/ingestion/source/dataplex/dataplex_helpers.py

```diff
@@ -0,0 +1,327 @@
+"""Helper functions and utilities for Dataplex source."""
+
+from collections import namedtuple
+from typing import Dict, Optional
+
+from google.api_core import exceptions
+from google.cloud import dataplex_v1
+
+import datahub.emitter.mce_builder as builder
+from datahub.emitter.mcp_builder import ProjectIdKey
+from datahub.metadata.schema_classes import (
+    ArrayTypeClass,
+    BooleanTypeClass,
+    BytesTypeClass,
+    DateTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    RecordTypeClass,
+    SchemaFieldDataTypeClass,
+    StringTypeClass,
+    TimeTypeClass,
+)
+
+# Type mapping from Dataplex to DataHub schema types
+DATAPLEX_TYPE_MAPPING = {
+    "BOOL": BooleanTypeClass(),
+    "BYTE": BytesTypeClass(),
+    "INT16": NumberTypeClass(),
+    "INT32": NumberTypeClass(),
+    "INT64": NumberTypeClass(),
+    "FLOAT": NumberTypeClass(),
+    "DOUBLE": NumberTypeClass(),
+    "DECIMAL": NumberTypeClass(),
+    "STRING": StringTypeClass(),
+    "BINARY": BytesTypeClass(),
+    "TIMESTAMP": TimeTypeClass(),
+    "DATE": DateTypeClass(),
+    "TIME": TimeTypeClass(),
+    "RECORD": RecordTypeClass(),
+    "NULL": NullTypeClass(),
+}
+
+
+# Container Key classes for Dataplex hierarchy
+class DataplexLakeKey(ProjectIdKey):
+    """Container key for Dataplex Lake."""
+
+    lake_id: str
+
+
+class DataplexZoneKey(DataplexLakeKey):
+    """Container key for Dataplex Zone (sub-container of Lake)."""
+
+    zone_id: str
+
+
+class DataplexAssetKey(DataplexZoneKey):
+    """Container key for Dataplex Asset (sub-container of Zone)."""
+
+    asset_id: str
+
+
+EntityDataTuple = namedtuple(
+    "EntityDataTuple",
+    ["lake_id", "zone_id", "entity_id", "asset_id", "source_platform", "dataset_id"],
+)
+
+
+def make_project_container_key(
```

Comment:
the make project container key should exist in some mcp helper file 🤔 

it's fine to have a wrapper method here but the implementation may refer to the one in mcp helper, if any 😅